help_function/src/onedpl_test_group_load.cpp

```diff
@@ -0,0 +1,114 @@
+    }
+  }
+
+  std::cout << func_name <<" pass\n";
+  return true;
+}
+
+bool test_load_blocked() {
+  sycl::queue q;
+  int data[512];
+  for (int i = 0; i < 512; i++) data[i] = i;
+
+  sycl::buffer<int, 1> buffer(data, 512);
+  q.submit([&](sycl::handler &h) {
+    using workgroup_load = dpct::group::workgroup_load<128, BLOCK_LOAD_DIRECT, int>;
+    size_t temp_storage_size = workgroup_load::get_local_memory_size(128);
+    sycl::local_accessor<uint8_t, 1> tacc(sycl::range<1>(temp_storage_size), h);
+    sycl::accessor data_accessor(buffer, h, sycl::read_write);
+    h.parallel_for(
+        sycl::nd_range<3>(sycl::range<3>(1, 1, 128), sycl::range<3>(1, 1, 128)),
+        [=](sycl::nd_item<3> item) {
+          int thread_data[4];
+          auto *d = data_accessor.get_multi_ptr<sycl::access::decorated::yes>().get();
+          auto *tmp = tacc.get_multi_ptr<sycl::access::decorated::yes>().get();
+          group_load(tmp).load(item,item.get_local_linear_id(), d, thread_data);
+        });
+  });
+  q.wait_and_throw();
+
+  sycl::host_accessor data_accessor(buffer, sycl::read_only);
+  const int *ptr = data_accessor.get_multi_ptr<sycl::access::decorated::yes>();
+  return helper_validation_function(ptr,"test_load_blocked");
+}
+
+bool test_load_striped() {
+  sycl::queue q;
+  int data[512];
+  for (int i = 0; i < 512; i++) data[i] = i;
+
+  sycl::buffer<int, 1> buffer(data, 512);
+  q.submit([&](sycl::handler &h) {
+    using workgroup_load = dpct::group::workgroup_load<128, BLOCK_LOAD_STRIPED, int>;
+    size_t temp_storage_size = workgroup_load::get_local_memory_size(128);
+    sycl::local_accessor<uint8_t, 1> tacc(sycl::range<1>(temp_storage_size), h);
+    sycl::accessor data_accessor(buffer, h, sycl::read_write);
+    h.parallel_for(
+        sycl::nd_range<3>(sycl::range<3>(1, 1, 128), sycl::range<3>(1, 1, 128)),
+        [=](sycl::nd_item<3> item) {
+          int thread_data[4];
+          auto *d = data_accessor.get_multi_ptr<sycl::access::decorated::yes>().get();
+          auto *tmp = tacc.get_multi_ptr<sycl::access::decorated::yes>().get();
+          group_load(tmp).load(item, d, thread_data);
+        });
+  });
+  q.wait_and_throw();
+
+  sycl::host_accessor data_accessor(buffer, sycl::read_only);
+  const int *ptr = data_accessor.get_multi_ptr<sycl::access::decorated::yes>();
+  return helper_validation_function(ptr,"test_load_blocked");
+}
+
+bool test_load_subgroup_striped() {
+  sycl::queue q;
+  int data[512];
+  for (int i = 0; i < 512; i++) data[i] = i;
+
+  sycl::buffer<int, 1> buffer(data, 512);
+  q.submit([&](sycl::handler &h) {
+    sycl::accessor dacc(buffer, h, sycl::read_write);
+    h.parallel_for(
+        sycl::nd_range<3>(sycl::range<3>(1, 1, 128), sycl::range<3>(1, 1, 128)),
+        [=](sycl::nd_item<3> item) {
+          int thread_data[4];
+          auto *d = dacc.get_multi_ptr<sycl::access::decorated::yes>().get();
+          dpct::group::uninitialized_load_subgroup_striped<128>(item, d, thread_data);

```

Comment:
We should similarly test the other free functions for correctness from my perspective even though they are already being indirectly invoked through the `workgroup_load` class API.

Reply:
Yes added the standalone function tests